fix #96971 suppress benign make release warnings

ericfont opened the pull request:
This commit cleans some -Wempty-body, -Wmaybe-uninitialized, -Wunused-variable, and -Wstrict-overflow compiler warnings that arise on my x86-64 and ARMv7-A arch linux machines when compiling release (not debug).
These compiler messages all don't seem to cause any bugs, but since they pollute the build output, they make it harder to spot potentially important warnings that might arise, so I'm making these small changes to remove or suppress them.
The "empty-body" warnings occur because Q_ASSERT statements are removed when QT_NO_DEBUG is defined, causing the else blocks to have nothing.  Surrounding the Q_ASSERT with brackets will suppress this warning.
The "maybe-uninitialized" warnings are all suppressed by assigning the variables to 0 either at initialization or in a switch default block.
The "unused-variable" warning is due to PeriodItem updatePeriods[] in preferences.cpp being defined but never used, and so I've commented out with #if 0 / #endif, so it is atleast still in the code.
The "strict-overflow" warning is due to compiler wanting to perform an optimization which would cause signed overflow.  However since I don't think any reasonable human use would be able to cause an overflow in findMaxOccupiedVoiceInBar(), I've surround the function with GCC pragmas which will suppress the warning.

ericfont commented:
I had a couple typos in my commit message, and I've set nmag = 0.0; instead of nmag = 0, since it is a double.

lasconic commented:
I'm not a big fan of the pragma. The rest looks good. PeriodItem can just be removed I think.

ericfont commented:
ok, I've removed the lines for PeriodItem.  And I removed the pragma lines (I'll just have to tolerate the compiler complaints about possible overflow in findMaxOccupiedVoiceInBar), but atleast my compile output is cleaned of everything else.

ericfont commented:
Zack's PR you just commited introduced another unused variable warning at this line: https://github.com/musescore/MuseScore/commit/0e05f08150fa0f6e154f3c4315d5e29eac3da69c#diff-af9d8f4883a1a78f5168a17809fb6129R3432

with this message:
<pre>/home/e/MuseScore/mscore/importmxmlpass2.cpp: In member function 'void Ms::MusicXMLParserPass2::clef(const QString&, Ms::Measure*, int)':
/home/e/MuseScore/mscore/importmxmlpass2.cpp:3432:11: warning: unused variable 'staves' [-Wunused-variable]
       int staves = part->nstaves();</pre>

I'm going to update this PR to delete that "int staves".

ericfont commented:
I figured out a way (without using pragmas or modifying compiler warning level) for the compiler to not complain about the possible signed overflow, and recommitted my PR.  The line that was making the compiler unhappy was:

https://github.com/ericfont/MuseScore/commit/9543fa4c66a9f6883eef14f48698773678afae50#diff-26f26b1a491494c21428173ddfce1256L713

and I've made it happy by casting those bar indexes from int's (as they are stored in their MidiChord class) into unsigned ints for this comparison.  For consistency in this function, I've made the other compares and unsigned.  Now I do notice that MidiChord's default constructor will initialize chord.barIndex = -1, which I assume represents an undefined state.  But I would think that by the time findMaxOccupiedVoiceInBar() is called, that those chords being compared would have a valid non-negative barIndex.  Maybe to be cautious, I could add a Q_ASSERT before those casts to make sure barIndex >= 0.  I haven't looked at code outside of findMaxOccupiedVoiceInBar(), so I don't know if that is necessary.  I'm attaching my build output indicating no warnings after I've made this change:  
[clean_make_release_build_output_archlinux_no_warnings_commit9543fa4.txt](https://github.com/musescore/MuseScore/files/127384/clean_make_release_build_output_archlinux_no_warnings_commit9543fa4.txt)

ericfont commented:
repushed due to a slight error I might have introduced if barIndex == 0 that I've fixed here: https://github.com/ericfont/MuseScore/commit/f111ee714649e1f8baeb416ee93c0c0fe21d054c#diff-26f26b1a491494c21428173ddfce1256R721

lasconic commented:
Tests are failing, not sure why. I will relaunch them.

* For periodItem you can also remove the struct definition just above.

lasconic commented:
OK tests are passing. So just the periodItem and good to merge.

ericfont commented:
looks like it is completed and passed now...was it the push prior to when I said "repushed due to a slight error I might have introduced if barIndex == 0" that was failing?  If so could have been because what I wrote in that previous mistaken push would have produced infinite look if barIndex=0.

ericfont commented:
ok...let me fix periodItem.
